components/notebook-controller/Dockerfile

```diff
@@ -11,6 +11,8 @@ ARG GOLANG_VERSION=1.21
 
 # Use ubi8/go-toolset as base image
 FROM registry.access.redhat.com/ubi8/go-toolset:${GOLANG_VERSION} as builder
+ARG TARGETOS
+ARG TARGETARCH
```

Comment:
> Should we have the TARGETARCH defined in our Makefile?

In general, I think not, or not yet. When we want to do

* https://github.com/opendatahub-io/kubeflow/issues/376

we can do it.

The way kubebuilder does it is something like https://github.com/kubeflow/notebooks/blob/notebooks-v2/workspaces/controller/Makefile#L106-L115 We'd have to do this with podman instead of docker buildx, but the approach is not very different. And we'd have to take into account osbs/konflux, which (I guess) builds every arch natively and then does the manifest in a separate step.